airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2768,11 +2772,27 @@ def prepare_template(self) -> None:
             with open(self.configuration) as file:
                 self.configuration = json.loads(file.read())
 
+    def _add_job_labels(self) -> None:
+        dag_label = self.dag_id.lower()
+        task_label = self.task_id.lower()
+
+        if len(dag_label) <= LABEL_SIZE_LIMIT and len(task_label) <= LABEL_SIZE_LIMIT:
+            if LABEL_REGEX.match(dag_label) and LABEL_REGEX.match(task_label):
+                if "labels" in self.configuration:
+                    if isinstance(self.configuration["labels"], dict):
```

Comment:
These checks are too naive—I would instead do something like

```python
try:
    self.configuration["labels"].upgrade({"airflow-dag": dag_label, "airflow-task": task_label})
except (KeyError, TypeError, ValueError):
    self.configuration["labels"] = {"airflow-dag": dag_label, "airflow-task": task_label}
```

Reply:
I've switched to `update` for brevity, but I'm not keen on coding by exception so I'm still checking I have a dict first.  This also means it respects a `"labels": None` value.